### ISSUE TITLE
Post events to FEBE endpoint

### ISSUE THREAD

**Issue by jamesjwarren** [thisissoon/FM-Perceptor#1]
The FEBE now provides an endpoint `POST /api/events` to receive events. https://github.com/thisissoon/FM-Frontend-API/pull/35

Accepts data in the following format:

``` json
{
  "_event": "queue.add",
  "uri": 1
}
```

Requests must be HMAC verified with a trusted client key.

Obvs the FEBE isn't live yet and there are still a few things we need to do before it can be put into the infrastructure, but would be cool to test this out when there is an opportunity.

**Comment by krak3n** [comment]
n1 :+1: 

Fancy having a Go :wink: at adding it into Perceptor?

**Comment by jamesjwarren** [comment]
Yeh that would be cool :smile:
Would it be implemented in `events` in place of redis, or would it need its own module?

**Comment by krak3n** [comment]
@jamesjwarren actually the `add` event is an event that is required by the player and is not one that Perceptor would forward onto the FE clients via the FE WS Service.

I think it's actually going to be the `FM-API` that needs to PUSH the `add` event to Peceptor. The flow would actually need to be:

```
FE > FEBE > API > Perceptor > Player (the add event is used to unblock queues)
                > FEBE > FE (Tell the other client a new track was added to the queue)
```

So what Perceptor really needs is a new REST endpoint for `add` events than then broadcasts the add event to player. The API would implement this endpoint when a new track is added to the playlist and concurrently send the `add` event to the WS server to tell other clients about the event.

Does that make sense?

**Comment by jamesjwarren** [comment]
Ah yes that makes sense. What about other events? I guess `add` `pause` `resume` `volume` `mute` would all come from the API and be PUSHed to Perceptor (as you described)? But `play` and `end` would surely need to come from perceptor and be pushed from there to the WS server?

**Comment by krak3n** [comment]
@jamesjwarren `add` is different to the other events since the player dosen't really take action on it like the others, for example a `pause` event needs the player to take action on that event and then feedback that it succeeded. So for something like pause the flow would be:

```
FE > FEBE > API > Perceptor (pause) > Player (pauses) > Perceptor (paused) > FEBE > FE
```

The `add` event is just used on the player to unblock the queue check so we don't poll the API every second.

So for the following we do need Perceptor to PUSH to the FEBE WS Server:
- `play`
- `end`
- `pause`
- `resume`
- `volume`
- `mute` 

Basically everything thats in `/events/*.go` that currently does a `Publish` to Redis needs a REST endpoint in `FEBE` to notify other FE clients.

**Comment by jamesjwarren** [comment]
Ok, thanks for clearing that up. So all those events can be implemented with the above FEBE endpoint with just `add` requiring the new REST endpoint in Perceptor and feedback from `FM-API` > `FEBE`. Will look into adding this to Perceptor, when I get a chance.
